Libs/Widgets/ctkPushButton.cpp

```diff
@@ -331,6 +351,7 @@ void ctkPushButton::paintEvent(QPaintEvent * _event)
   // all the computations have been made infering the text would be left oriented
   tf &= ~Qt::AlignHCenter & ~Qt::AlignRight;
   tf |= Qt::AlignLeft;
+  QString elidedText = opt.fontMetrics.elidedText(opt.text, d->ElideMode, opt.rect.width());
```

Comment:
does it work with and without icons ?

Reply:
Tested now with all icon and text alignment combinations. There were a couple of bugs (they did not seem to be related to the eliding but fixed them anyway).